test/k8sT/Services.go

```diff
@@ -445,8 +445,27 @@ var _ = Describe("K8sServicesTest", func() {
 					doRequests("http://"+lbIP, 10, helpers.K8s2)
 				})
 			})
-		})
 
+			It("Tests with direct routing and DSR", func() {
+				deleteCiliumDS(kubectl)
+				DeployCiliumOptionsAndDNS(kubectl, []string{
+					"--set global.nodePort.enabled=true",
+					"--set global.nodePort.device=" + nativeDev,
+					"--set global.nodePort.dsr=true",
+					"--set global.tunnel=disabled",
+					"--set global.autoDirectNodeRoutes=true",
+					"--set global.ipv6.enabled=false",
+				})
+
+				var data v1.Service
+				err := kubectl.Get(helpers.DefaultNamespace, "service test-nodeport").Unmarshal(&data)
+				Expect(err).Should(BeNil(), "Can not retrieve service")
+				url := getURL(helpers.K8s1Ip, data.Spec.Ports[0].NodePort)
+				doRequestsFromOutsideClient(url, 10, true)
+				url = getURL(helpers.K8s2Ip, data.Spec.Ports[0].NodePort)
+				doRequestsFromOutsideClient(url, 10, true)
+			})
+		})
```

Comment:
How is this test checking the traffic was directly sent from the destination node, i.e. the backend, to the client?

Reply:
It doesn't. I don't see any reasonably non-complex way to test this.